### PR TITLE
Feat/new signtx process

### PR DESCRIPTION
**Pull request by szymonlesisz**

### CODE REVIEW

File: src/js/backend/BlockchainLink.js [hunk 1]
```diff
@@ -112,23 +109,12 @@ export default class Blockchain {
         }
     }
 
-    async loadTransaction(id: string) {
-        const transaction = await this.link.getTransaction(id);
-        return BitcoinJsTransaction.fromHex(transaction.tx.hex, this.coinInfo.network);
-    }
-
```
Comment (mroz22) [inline src/js/backend/BlockchainLink.js:119]: this is just some cleanup of unused methods right?

Comment (szymonlesisz) [reply src/js/backend/BlockchainLink.js:119]: right, BitcoinJsTransaction logic is moved to `refTxs` utility.
its not need here, in blockchainlink file

**Comment by mroz22** [comment]
:scream: 121 changed files :scream: 
should be rebased I guess?

**Comment by szymonlesisz** [comment]
yup, rebased

File: src/js/core/methods/helpers/signtx.js [hunk 2]
```diff
@@ -5,202 +5,193 @@ import type { RefTransaction, TransactionOptions, SignedTransaction } from '../.
...
+    if (request_type === 'TXOUTPUT') {
+        if (!tx.bin_outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXOUTPUT: ${tx_hash}`);
+        return typedCall('TxAckPrevOutput', 'TxRequest', { tx: { output: tx.bin_outputs[details.request_index] } });
+    }
+    if (request_type === 'TXORIGINPUT') {
+        if (!tx.outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXORIGINPUT: ${tx_hash}`);
+        return typedCall('TxAckInput', 'TxRequest', { tx: { input: tx.inputs[details.request_index] } });
+    }
+    if (request_type === 'TXORIGOUTPUT' && tx.outputs) {
+        if (!tx.outputs) throw ERRORS.TypedError('Runtime', `requestPrevTxInfo: Requested unknown TXORIGOUTPUT: ${tx_hash}`);
```
Comment (mroz22) [inline src/js/core/methods/helpers/signtx.js:55]: this is weird, you already have && tx.outputs on line 50. is that expected?

Comment (szymonlesisz) [reply src/js/core/methods/helpers/signtx.js:55]: good catch, it shouldn't be in line 50

Comment (szymonlesisz) [reply src/js/core/methods/helpers/signtx.js:55]: addressed 9e1c7a376ff627be22f3fd2d690faeec7eccfffb

File: src/js/core/methods/SignTransaction.js [hunk 3]
```diff
@@ -122,22 +128,43 @@ export default class SignTransaction extends AbstractMethod {
...
+                    let addresses = params.addresses;
+                    // sender account addresses not provided
+                    // fetch account info from the blockbook
+                    if (!addresses) {
+                        // TODO: validate inputs address_n's === same account
+                        const node = await device.getCommands().getHDNode(params.inputs[0].address_n.slice(0, 3), params.coinInfo);
+                        const account = await blockchain.getAccountInfo({
+                            descriptor: node.xpubSegwit || node.xpub,
+                            coin: params.coinInfo.name,
+                            details: 'tokens',
```
Comment (mroz22) [inline src/js/core/methods/SignTransaction.js:154]: just out of curiosity. is 'tokens' needed here? wouldnt 'basic' suffice?

Comment (mroz22) [reply src/js/core/methods/SignTransaction.js:154]: pulling respective part of documentation here: 
 - `basic` (default) return only account balances, without any derived addresses or transaction history
 - `tokens` - response with derived addresses (Bitcoin-like accounts) and ERC20 tokens (Ethereum-like accounts), subject of `tokens` param
  - `tokenBalances` - same as `tokens` with balances, subject of `tokens` param
  - `txs` - `tokenBalances` + complete account transaction history

**Review by keraf — APPROVED** [review]

File: src/js/core/methods/SignTransaction.js (hunk 3 above)
Comment (szymonlesisz) [reply src/js/core/methods/SignTransaction.js:154]: basic will not return derived addresses and addresses is what we want here :)

Comment (mroz22) [reply src/js/core/methods/SignTransaction.js:154]: Ah, I got confused by the "tokens" keyword

**Review by mroz22 — APPROVED** [review]

**Comment by keraf** [comment]
Squash and good to go :slightly_smiling_face:

**Pull request merged**